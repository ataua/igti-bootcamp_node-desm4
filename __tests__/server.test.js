const express = require('express');
const request = require('supertest');
const db = require('../src/db');

const app = express().use(express.json()).use('/', require('../src/app'));

test('Servidor na porta 5678', async () => {
  const resposta = await request(app).get('/');

  expect(resposta.status).toBe(200);
});

describe('Testes do desafio M4', () => {
  beforeEach(async () => {
    await db.produto.destroy({ where: {} });
  });

  const produtoTeste = {
    Codigo: 'DESM4-12345',
    Descricao: 'amarelo',
    Preco: 123.45,
  };

  const produtoTeste2 = {
    Codigo: 'DESM4-12345',
    Descricao: 'azul',
    Preco: 45.67,
  };

  test('Cria produto corretamente', async () => {
    const resposta = await request(app)
      .post('/produto')
      .send(produtoTeste);

    const produto = await db.produto.findOne({ where: { Codigo: produtoTeste.Codigo } });
    delete produto.createdAt;
    delete produto.updatedAt;

    expect(resposta.body).not.toBeUndefined();
    expect(resposta.status).toBe(201);
    expect(produto.Preco).toBe(produtoTeste.Preco);
  });

  test('Busca a lista de produtos', async () => {
    await db.produto.create(produtoTeste);
    const resposta = await request(app).get('/produto');
    expect(resposta.body.length).toBe(1);
    expect(resposta.body[0].Preco).toBe(produtoTeste.Preco);
    expect(resposta.status).toBe(200);
  });

  test('Deleta produto', async () => {
    const { Codigo } = produtoTeste;
    await db.produto.create(produtoTeste);
    const salvo = db.produto.findOne({ where: { Codigo } });
    expect(salvo).not.toBeUndefined();

    const result = await request(app).delete(`/produto/${produtoTeste.Codigo}`);
    expect(result.status).toBe(200);
    expect(result.body.mensagem).not.toBeUndefined();
  });

  test('Atualiza produto', async () => {
    await db.produto.create(produtoTeste);
    const result = await request(app)
      .put('/produto')
      .send(produtoTeste2);

    expect(result.body.Preco).toBe(produtoTeste2.Preco);
  });
});
