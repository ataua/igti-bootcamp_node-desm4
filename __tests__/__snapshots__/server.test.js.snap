// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes do desafio M4 Cria produto corretamente 1`] = `
Object {
  "Codigo": "DESM4-12345",
  "Descricao": "amarelo",
  "Preco": 123.45,
  "_changed": Set {},
  "_options": Object {
    "_schema": null,
    "_schemaDelimiter": "",
    "attributes": Array [
      "Codigo",
      "Descricao",
      "Preco",
      "createdAt",
      "updatedAt",
    ],
    "isNewRecord": false,
    "raw": true,
  },
  "_previousDataValues": Object {
    "Codigo": "DESM4-12345",
    "Descricao": "amarelo",
    "Preco": 123.45,
    "createdAt": 2023-03-27T23:27:28.354Z,
    "updatedAt": 2023-03-27T23:27:28.354Z,
  },
  "dataValues": Object {
    "Codigo": "DESM4-12345",
    "Descricao": "amarelo",
    "Preco": 123.45,
    "createdAt": 2023-03-27T23:27:28.354Z,
    "updatedAt": 2023-03-27T23:27:28.354Z,
  },
  "isNewRecord": false,
  "uniqno": 1,
}
`;
